src/mono/mono/mini/CMakeLists.txt

```diff
@@ -541,14 +538,14 @@ if(NOT DISABLE_EXECUTABLES)
       set_target_properties(mono-sgen PROPERTIES INSTALL_RPATH "$ORIGIN")
     endif()
   endif()
-  target_link_libraries(mono-sgen PRIVATE monoapi eglib_api monosgen-static)
+  target_link_libraries(mono-sgen PRIVATE monoapi eglib_api monosgen-static zlibstatic)
   if (HOST_WASM)
-    target_link_libraries(mono-sgen PRIVATE mono-wasm-nosimd)
+    target_link_libraries(mono-sgen PRIVATE mono-wasm-nosimd zlibstatic)
```

Comment:
we don't need this here, it is already done by the previous target_link_libraries